webpack/assets/javascripts/pages/statistics_page.jsx

```diff
@@ -0,0 +1,43 @@
+import donut from '../charts/c3DonutChartService';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import StatisticsChartsList from "../react_app/components/charts/StatisticsChartsList";
+
+export function generateChart(details) {
+    let dataEventHandlers = {
+        onclick: getClickHandler(details.url)
+    };
+
+    let config = donut.getDonutConfig(details.data, '#' + details.id + 'Chart', dataEventHandlers);
+
+    donut.generate(config);
+
+    function getClickHandler(url) {
+        return function (d, i) {
+            var val = '"' + encodeURIComponent(d.name) + '"';
+            if (url.indexOf("~VAL1~") != -1) {
+                url = url.replace("~VAL1~", val);
+            } else {
+                if (val.indexOf(" ") != -1) val = '"' + val + '"';
+                url = url.replace("~VAL~", val);
+            }
+            window.location.href = url;
+        };
+    }
+}
+
+export function generateModalChart(details) {
+    let config = donut.getLargeDonutConfig(details.data, '#' + details.id + 'ModalChart');
+    donut.generate(config);
+}
+
+// TODO: consider splitting off  this function which needs to be code not packaged be webpack,
+// which is not a requiremenct for the other exported functions
+export function documentReady() {
+    let reactNode = document.getElementById('statistics');
```

Comment:
does it make sense to pass the div name as a parameter instead as a hard coded value here?